### PR TITLE
Add more `@safe` annotations

### PR DESCRIPTION
**Pull request by dkorpel**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: compiler/src/dmd/funcsem.d [hunk 1]
```diff
@@ -2912,6 +2912,7 @@ bool setUnsafePreview(Scope* sc, FeatureState fs, bool gag, Loc loc, const(char)
         {
             import dmd.func : AttributeViolation;
             sc.func.safetyViolation = new AttributeViolation(loc, msg, arg0, arg1, arg2);
+            sc.func.hasSafeViolations = true;
```
Comment (thewilsonator) [inline compiler/src/dmd/funcsem.d:2915]:
```
compiler/src/dmd/funcsem.d(2915,20): Error: no property `hasSafeViolations` for `(*sc).func` of type `dmd.func.FuncDeclaration`
compiler/src/dmd/func.d(193,14):        class `FuncDeclaration` defined here
```
assuming this is intentional, please leave it for a followup PR.

Comment (dkorpel) [reply compiler/src/dmd/funcsem.d]: Whoops, that wasn't supposed to be part of this commit

File: compiler/src/dmd/backend/debugprint.d [hunk 2]
```diff
@@ -328,7 +328,6 @@ void WRdefnod(ref GlobalOptimizer go)
   }
 }
 
-@trusted
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/debugprint.d:331]: any reason for not marking these `@safe`?

File: compiler/src/dmd/backend/ty.d [hunk 3]
```diff
@@ -287,7 +287,6 @@ bool tyvector(tym_t ty) { return tybasic(ty) >= TYfloat4 && tybasic(ty) <= TYull
 uint tyshort(tym_t ty) { return tytab[ty & 0xFF] & TYFLshort; }
 
 /* Detect TYlong or TYulong     */
-@trusted
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/ty.d:290]: ditto

File: compiler/src/dmd/backend/x86/cgcod.d [hunk 4]
```diff
@@ -2384,7 +2384,6 @@ private void loadcse(ref CodeBuilder cdb,elem *e,reg_t reg,regm_t regm)
 }
 
 
-@trusted
```
Comment (thewilsonator) [inline compiler/src/dmd/backend/x86/cgcod.d:2387]: ditto

**Review by thewilsonator — COMMENTED** [review]
I agree they shouldn't be trusted, being so trivially safe.

File: compiler/src/dmd/dsymbol.d [hunk 5]
```diff
@@ -851,7 +851,7 @@ extern (C++) class Dsymbol : ASTNode
      */
     extern (D) static bool oneMembers(Dsymbols* members, out Dsymbol ps, Identifier ident)
     {
-        //printf("Dsymbol::oneMembers() %d\n", members ? members.length : 0);
+        printf("Dsymbol::oneMembers() %d\n", members ? cast(int) members.length : 0);
```
Comment (thewilsonator) [inline compiler/src/dmd/dsymbol.d:854]: don't forget to turn this `printf` back off.

File: compiler/src/dmd/backend/debugprint.d (hunk 2 above)
Comment (dkorpel) [reply compiler/src/dmd/backend/debugprint.d:331]: There's `@safe:` on top of the file

File: compiler/src/dmd/dsymbol.d (hunk 5 above)
Comment (dkorpel) [reply compiler/src/dmd/dsymbol.d]: I really should commit before constantly switching branches :sweat_smile:

**Comment by thewilsonator** [comment]
```
src/dmd/denum.d(67): Deprecation: `dmd.denum.EnumDeclaration.this` has stricter attributes than its destructor (`@system`)
src/dmd/denum.d(67):        The destructor will be called if an exception is thrown
src/dmd/denum.d(67):        Either make the constructor `nothrow` or adjust the field destructors
src/dmd/nspace.d(65): Deprecation: `dmd.nspace.Nspace.this` has stricter attributes than its destructor (`@system`)
src/dmd/nspace.d(65):        The destructor will be called if an exception is thrown
src/dmd/nspace.d(65):        Either make the constructor `nothrow` or adjust the field destructors
```

File: compiler/src/dmd/backend/debugprint.d (hunk 2 above)
Comment (RazvanN7) [reply compiler/src/dmd/backend/debugprint.d:331]: Ha! On-mass attributes here to bite you again!

**Pull request merged**